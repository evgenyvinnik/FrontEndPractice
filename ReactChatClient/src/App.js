import React, {useEffect, useState} from 'react';
import axios from 'axios';

import './App.css';
import 'amazon-connect-chatjs';

function App() {
  let AmazonConnect = window.connect;
  const [chatResult, setChatResult] = useState({});
  const [session, setSession] = useState();


  const [inputValue, setInputValue] = useState('');
  const [receivedMessages, setReceivedMessages] = useState([]);

  const apiGatewayEndpoint = 'https://ntdtxbt3vf.execute-api.us-east-1.amazonaws.com/Prod/'
  var initiateChatRequest = {
    ParticipantDetails: {
        DisplayName: 'AdrianCustomer'
    },
    ContactFlowId: '8b87d0de-9883-4745-9479-0abf68801bc8',
    InstanceId: 'ac9744f5-ee87-404c-85b6-40bacee5a018'
  };

  useEffect(() => {
    console.log('Setting global config');
    AmazonConnect.ChatSession.setGlobalConfig({
      region: "us-east-1"
    });
    console.log('global config set');
  },[AmazonConnect.ChatSession]);


  useEffect(() => {
    console.log('Requesting initiate chat');

    axios.post(apiGatewayEndpoint, initiateChatRequest).then(res => {
        console.log('Start Chat Result');
        console.log(res.data.data.startChatResult);
        setChatResult(res.data.data.startChatResult);
        setSession(AmazonConnect.ChatSession.create({
          chatDetails: res.data.data.startChatResult,
          type: 'CUSTOMER'
        }));
    })
  }, []);

  useEffect(() => {
    if(session) {
      console.log('Test');
      session.connect().then((response) => {
        console.log("successful connection: " + JSON.stringify(response));
      }, (error) => {
        console.log("unsuccessful connection " + JSON.stringify(error));
      });
      session.onMessage((message) => {
        console.log("Received message: " + JSON.stringify(message));

        setReceivedMessages(receivedMessages => [...receivedMessages, message.data.Content]);
      });
      session.onConnectionEstablished((data) => {
        setReceivedMessages(receivedMessages => [...receivedMessages, 'CONNECTION ESTABLISHED']);
      })
    }
  },[session]);

  const handleClick = () => {
    session.controller.sendMessage({
      message: inputValue,
      contentType: "text/plain"
    })
  }

  const handleInput = (e) => {
    setInputValue(e.target.value);
  }

console.log('receivedMessages', receivedMessages);
  return (
    <div>
      <header>
        <h1>Amazon Connect - REACT Test</h1>
      </header>
      <br></br>
      <br></br>
      <div>
        Test area for chat:
        <ul>
        {receivedMessages.map(message => {
          return <li>{message}</li>;
        })}
        </ul>
      </div>
      <input onChange={handleInput}></input>
      <button onClick={handleClick}>Send</button>
    </div>
  );
}

export default App;
