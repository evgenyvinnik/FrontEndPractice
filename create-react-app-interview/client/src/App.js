import React, { useState } from "react";

const App = () => {
  const [users, setUsers] = useState([]);

  const fetchUsers = async () => {
    const response = await fetch("/users/");
    const data = await response.json();
    return data;
  }

  //useEffect(addFavorite, [])

  const getUsers = async () => {
    const retrievedUsers = await fetchUsers();
    console.log(retrievedUsers);
    setUsers(retrievedUsers);
  }

  return (
    <div>
      <button onClick={getUsers}>Fetch Users</button>
    </div>
  )
}

export default App;
