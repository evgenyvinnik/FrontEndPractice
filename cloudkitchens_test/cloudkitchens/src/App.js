import React, { useEffect, useState} from 'react';
import io from "socket.io-client";
import './App.css';

function App() {
  const [socket, setSocket] = useState(null);
  const [orders, setOrders] = useState({});
  // const [filter, setFilter] = useState(undefined);
  const [pageFilter, setPageFilter] = useState(0);
  // const [filtered, filterOrders] = useState({});
 
  // establish socket connection
  useEffect(() => {
    setSocket(io('http://localhost:4000'));
  }, []);

  useEffect(() => {
    console.log("pageFilter value" + pageFilter);
  }, [pageFilter]);

 
  // subscribe to the socket event
  useEffect(() => {
    if (!socket) return;
 
    socket.on('connect', () => {
      socket.emit('order_event');
    });

    socket.on('order_event', (data) => {
      var newOrderData = {};
      data.forEach((orderData) => {
          newOrderData = {...newOrderData, [orderData.id]: orderData};
      });

      setOrders({...orders, ...newOrderData});
    });
  }, [socket, orders]);

  function page(pageId){
    var page = {};
    const pageSize = 10;

    console.log("pageId" +  pageId);
    console.log("pageId" +  pageId);
    page =  Object.values(orders).slice(pageId * pageSize, pageId * pageSize + pageSize);
    // page 0
    // slice from 0, 10
    // page 1
    // slice from 10, 20
    // page 2
    // slice from 20 to 30

    console.log("page length" +  Object.values(page).length);

    return Object.values(page)
    .map((order) => {
      return (
          <tr key={order.id}>
            <td>{order.id}</td>
            <td>{order.customer}</td>
            <td>{order.item}</td>
            <td>{order.price}</td>
            <td>{order.event_name}</td>
          </tr>
      )
    });
  };
 
  return (
    <div className="app">
      <div className="search">
        {<input
          name="searchFilter"
          type="number"
          onChange={(e) => setPageFilter(e.target.value)}
          value={pageFilter}>
        </input>}
      </div>
      <div className="table">
        <table>
          <thead>
            <tr>
              <th>ID</th>
              <th>Customer</th>
              <th>Item</th>
              <th>Price</th>
              <th>Status</th>
            </tr>
          </thead>
          <tbody>
            {
              page(pageFilter)
            }
          </tbody>
        </table>
      </div>
    </div>
  );
}

export default App;
