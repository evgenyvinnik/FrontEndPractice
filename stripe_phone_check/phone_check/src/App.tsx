
import logo from './logo.svg';
import './App.css';
import {PhoneInput} from './components/PhoneInput';

function App() {
  return (
    <div className="App">
      <PhoneInput/>
    </div>
  );
}

export default App;
