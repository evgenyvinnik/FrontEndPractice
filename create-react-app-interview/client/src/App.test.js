import React from "react";
import ReactDOM from "react-dom";
import ReactTestUtils from "react-dom/test-utils";
import ReactTestRenderer from "react-test-renderer";
import * as ReactTestingLibrary from "@testing-library/react";
import * as Enzyme from "enzyme";
import App from "./App";

describe("App", () => {
  it("renders with `react-dom`", () => {
    const domContainer = document.createElement("div");
    expect(() => ReactDOM.render(<App />, domContainer)).not.toThrow();
  });

  it("renders with `react-dom/test-utils`", () => {
    const component = ReactTestUtils.renderIntoDocument(<App />);
    const heading = ReactTestUtils.findRenderedDOMComponentWithTag(
      component,
      "H1"
    );
    expect(heading.textContent).toBe("Hello World");
  });

  it("renders with `react-test-renderer`", () => {
    const component = ReactTestRenderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with `@testing-library/react", () => {
    const { getByText } = ReactTestingLibrary.render(<App />);
    const heading = getByText("Hello World");
    expect(heading.tagName).toBe("H1");
  });

  it("renders with `enzyme`", () => {
    const tree = Enzyme.shallow(<App />);
    expect(tree).toMatchSnapshot();
  });
});
