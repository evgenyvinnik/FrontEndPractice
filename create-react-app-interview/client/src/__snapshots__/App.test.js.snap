// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders with \`enzyme\` 1`] = `
<h1>
  Hello World
</h1>
`;

exports[`App renders with \`react-test-renderer\` 1`] = `
<h1>
  Hello World
</h1>
`;
